abstractions/common/modules.py

```diff
@@ -122,19 +122,33 @@ class InverseModel(torch.nn.Module):
             The size of the abstract state
         - num_actions : Int
             The number of actions in the environment
-    
-    Notes:
-        - This does not support continuous action spaces right now. TODO
+
     """
-    def __init__(self, args, feature_size, num_actions):
+    def __init__(self, args, feature_size, num_actions, discrete=False):
         super(InverseModel, self).__init__()
-        self.model = torch.nn.Sequential(torch.nn.Linear(feature_size * 2, args.hidden_size),
-                                         torch.nn.ReLU(),
-                                         torch.nn.Linear(args.hidden_size, num_actions))
+        self.discrete = discrete
+        self.body = torch.nn.Sequential(
+            torch.nn.Linear(feature_size * 2, args.hidden_size),
+            torch.nn.ReLU(),
+        )
+        if self.discrete:
+            self.log_pr_linear = torch.nn.Linear(args.hidden_size, num_actions)
+        else:
+            self.mean_linear = torch.nn.Linear(args.hidden_size, num_actions)
+            self.log_std_linear = torch.nn.Linear(args.hidden_size, num_actions)
 
     def forward(self, z0, z1):
         context = torch.cat((z0, z1), -1)
-        return self.model(context)
+        shared_vector = self.body(context)
+
+        if self.discrete:
+            return self.log_pr_linear(shared_vector)
+        else:
+            mean = self.mean_linear(shared_vector)
+            log_std = self.log_std_linear(shared_vector)
+            log_std = torch.clamp(log_std, min=LOG_SIG_MIN, max=LOG_SIG_MAX)
+            std = log_std.exp()
```

Comment:
Note that I left std as linear -> clamp -> exp, rather than linear -> softplus, because it looked like it had a smoother loss plot.

Reply:
Reasonable.